app/src/main/java/sdp/moneyrun/LeaderboardActivity.java

```diff
@@ -12,15 +15,27 @@
 
     private ArrayList<Player> playerList = new ArrayList<>();
     private LeaderboardListAdapter ldbAdapter;
+    private Button goBackButton;
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_leaderboard);
         addAdapter();
+        goBackButton = findViewById(R.id.leaderboard_go_back_button);
+        linkGoBackButton();
         //TODO
         // Put addPlayer with local cache
     }
 
+    private void linkGoBackButton(){
```

Comment:
OK I actually didn't know it worked that way

Reply:
I think it is better not to go back that way because Activities are piled on a stack. When you leave with the arrow go back button, it pops it from the stack but that button would just  fill the stack